--- a/proto/gnmi_ext/BUILD.bazel.orig	2025-11-06 10:00:38.700293000 +0200
+++ b/proto/gnmi_ext/BUILD.bazel	2025-11-06 10:00:56.958272398 +0200
@@ -1,48 +1,13 @@
-# Copyright 2021 Google LLC
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     https://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-# Supporting infrastructure for implementing and testing PINS.
-#
-
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
-load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-
-package(
-    default_visibility = ["//visibility:public"],
-    licenses = ["notice"],
-)
-
-proto_library(
-    name = "gnmi_ext_proto",
-    srcs = ["gnmi_ext.proto"],
-    deps = ["@com_google_protobuf//:duration_proto" ],
-    import_prefix = "github.com/openconfig/gnmi",
-)
-
-cc_proto_library(
-    name = "gnmi_ext_cc_proto",
-    deps = [":gnmi_ext_proto"],
-)
-
-go_proto_library(
-    name = "gnmi_ext_go_proto",
-    importpath = "github.com/openconfig/gnmi_ext",
-    proto = ":gnmi_ext_proto",
-)
 
 go_library(
     name = "gnmi_ext",
-    embed = [":gnmi_ext_go_proto"],
-    importpath = "github.com/openconfig/gnmi_ext",
+    srcs = ["gnmi_ext.pb.go"],
+    importpath = "github.com/openconfig/gnmi/proto/gnmi_ext",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@org_golang_google_protobuf//reflect/protoreflect",
+        "@org_golang_google_protobuf//runtime/protoimpl",
+        "@org_golang_google_protobuf//types/known/durationpb",
+    ],
 )
--- a/proto/gnmi/BUILD.bazel.orig	2025-11-06 10:00:38.701292998 +0200
+++ b/proto/gnmi/BUILD.bazel	2025-11-06 10:01:02.141266457 +0200
@@ -1,64 +1,21 @@
-# Copyright 2021 Google LLC
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     https://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-# Supporting infrastructure for implementing and testing PINS.
-#
-
-load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
-load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
-
-package(
-    default_visibility = ["//visibility:public"],
-    licenses = ["notice"],
-)
 
-proto_library(
-    name = "gnmi_proto",
-    srcs = ["gnmi.proto"],
-    import_prefix = "github.com/openconfig/gnmi",
-    deps = [
-        "//proto/gnmi_ext:gnmi_ext_proto",
-        "@com_google_protobuf//:any_proto",
-        "@com_google_protobuf//:descriptor_proto",
+go_library(
+    name = "gnmi",
+    srcs = [
+        "gnmi.pb.go",
+        "gnmi_grpc.pb.go",
     ],
-)
-
-cc_proto_library(
-    name = "gnmi_cc_proto",
-    deps = [":gnmi_proto"],
-)
-
-cc_grpc_library(
-    name = "gnmi_cc_grpc_proto",
-    srcs = [":gnmi_proto"],
-    generate_mocks = True,
-    grpc_only = True,
-    deps = [":gnmi_cc_proto"],
-)
-
-go_proto_library(
-    name = "gnmi_go_proto",
-    importpath = "github.com/openconfig/gnmi",
-    proto = ":gnmi_proto",
+    importpath = "github.com/openconfig/gnmi/proto/gnmi",
+    visibility = ["//visibility:public"],
     deps = [
-        "//proto/gnmi_ext:gnmi_ext_go_proto",
+        "//proto/gnmi_ext",
+        "@org_golang_google_grpc//:grpc",
+        "@org_golang_google_grpc//codes",
+        "@org_golang_google_grpc//status",
+        "@org_golang_google_protobuf//reflect/protoreflect",
+        "@org_golang_google_protobuf//runtime/protoimpl",
+        "@org_golang_google_protobuf//types/descriptorpb",
+        "@org_golang_google_protobuf//types/known/anypb",
     ],
 )
-
-go_library(
-    name = "gnmi",
-    embed = [":gnmi_go_proto"],
-    importpath = "github.com/openconfig/gnmi",
-)
--- /dev/null	2025-11-03 19:52:56.735552048 +0200
+++ b/value/BUILD.bazel	2025-11-06 10:01:02.387266174 +0200
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "value",
+    srcs = ["value.go"],
+    importpath = "github.com/openconfig/gnmi/value",
+    visibility = ["//visibility:public"],
+    deps = ["//proto/gnmi"],
+)
--- /dev/null	2025-11-03 19:52:56.735552048 +0200
+++ b/ctree/BUILD.bazel	2025-11-06 10:01:02.641265881 +0200
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "ctree",
+    srcs = ["tree.go"],
+    importpath = "github.com/openconfig/gnmi/ctree",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//proto/gnmi",
+        "@org_golang_google_protobuf//proto",
+    ],
+)
--- /dev/null	2025-11-03 19:52:56.735552048 +0200
+++ b/errdiff/BUILD.bazel	2025-11-06 10:01:02.952265523 +0200
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "errdiff",
+    srcs = ["errdiff.go"],
+    importpath = "github.com/openconfig/gnmi/errdiff",
+    visibility = ["//visibility:public"],
+)
--- /dev/null	2025-11-03 19:52:56.735552048 +0200
+++ b/errlist/BUILD.bazel	2025-11-06 10:01:03.217265218 +0200
@@ -0,0 +1,11 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "errlist",
+    srcs = ["errlist.go"],
+    importpath = "github.com/openconfig/gnmi/errlist",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@org_golang_x_exp//slices",
+    ],
+)
