diff --git a/README.md b/README.md
index df07ad8..3344cff 100644
--- a/README.md
+++ b/README.md
@@ -19,6 +19,7 @@ The forms include:
 
 *  tree - a simple tree representation
 *  types - list understood types extracted from the schema
+*  annotate - a template file to annotate the yang modules
 
 The yang package, and the goyang program, are not complete and are a work in
 progress.
diff --git a/pkg/yang/ast.go b/pkg/yang/ast.go
index 6fcf11e..4aea362 100644
--- a/pkg/yang/ast.go
+++ b/pkg/yang/ast.go
@@ -403,6 +403,11 @@ func initTypes(at reflect.Type, d *typeDictionary) {
 			descend(name, f.Type)
 
 			fn = func(s *Statement, v, p reflect.Value) error {
+				if s.Keyword == "yang-version" ||
+					s.Keyword == "import" {
+					// unset for optimization
+					return nil
+				}
 				if v.Type() != at {
 					panic(fmt.Sprintf("given type %s, need type %s", v.Type(), at))
 				}
diff --git a/pkg/yang/entry.go b/pkg/yang/entry.go
index 81b46c8..21d2139 100644
--- a/pkg/yang/entry.go
+++ b/pkg/yang/entry.go
@@ -30,6 +30,8 @@ import (
 	"sort"
 	"strconv"
 	"strings"
+	"sync"
+	"sync/atomic"
 
 	"github.com/openconfig/goyang/pkg/indent"
 )
@@ -71,6 +73,21 @@ type deviationPresence struct {
 	hasMaxElements bool
 }
 
+type PerformanceOpts struct {
+	// The ChildSchemaCache accelerates the ChildSchema function in the ygot
+	// library.  A bottleneck there is mapping a reflect package StructTag to a
+	// child schema (Entry struct in Entry's Dir map).  This cache saves the
+	// result of that mapping to go directly from the StructTag to the child
+	// Entry without having to do the string manipulation and map iteration.
+	ChildSchemaCache  map[reflect.StructTag]*Entry
+	ChildSchemaMutex  sync.RWMutex                 `json:"-"`
+
+	// The IsSchemaValidated allows us to perform the validation of list keys
+	// and string reg-ex once in the ygot library rather than every time the
+	// schema tree is traversed.
+	IsSchemaValidated atomic.Bool                  `json:"-"`
+}
+
 // Entry represents a single schema tree node, which can be a directory
 // (containing a subtree) or a leaf node (which contains YANG types that have
 // no children, e.g., leaf, leaf-list). They can be distinguished by whether
@@ -92,8 +109,9 @@ type Entry struct {
 	Mandatory   TriState  `json:",omitempty"` // whether this entry is mandatory in the tree
 
 	// Fields associated with directory nodes
-	Dir map[string]*Entry `json:",omitempty"`
-	Key string            `json:",omitempty"` // Optional key name for lists (i.e., maps)
+	Dir      map[string]*Entry `json:",omitempty"`
+	DirOKeys []string          // Ordered Keys list in Dir
+	Key      string            `json:",omitempty"` // Optional key name for lists (i.e., maps)
 
 	// Fields associated with leaf nodes
 	Type *YangType `json:",omitempty"`
@@ -133,6 +151,8 @@ type Entry struct {
 	// the augmenting entity per RFC6020 Section 7.15.2. The namespace
 	// of the Entry should be accessed using the Namespace function.
 	namespace *Value
+
+	PerfOpts  *PerformanceOpts `json:",omitempty"`
 }
 
 // An RPCEntry contains information related to an RPC Node.
@@ -289,11 +309,13 @@ func (k EntryKind) String() string {
 // newDirectory returns an empty directory Entry.
 func newDirectory(n Node) *Entry {
 	return &Entry{
-		Kind:  DirectoryEntry,
-		Dir:   make(map[string]*Entry),
-		Node:  n,
-		Name:  n.NName(),
-		Extra: map[string][]interface{}{},
+		Kind:     DirectoryEntry,
+		Dir:      make(map[string]*Entry),
+		DirOKeys: make([]string, 0),
+		Node:     n,
+		Name:     n.NName(),
+		Extra:    map[string][]interface{}{},
+		PerfOpts: &PerformanceOpts{},
 	}
 }
 
@@ -387,6 +409,7 @@ func (e *Entry) add(key string, value *Entry) *Entry {
 		return e
 	}
 	e.Dir[key] = value
+	e.DirOKeys = append(e.DirOKeys, key)
 	return e
 }
 
@@ -595,6 +618,8 @@ func ToEntry(n Node) (e *Entry) {
 		addExtraKeywordsToLeafEntry(n, e)
 		e.Mandatory, err = tristateValue(s.Mandatory)
 		e.addError(err)
+		e.Description = ""
+		e.PerfOpts = &PerformanceOpts{}
 		return e
 	case *LeafList:
 		// Create the equivalent leaf element that we are a list of.
@@ -627,6 +652,8 @@ func ToEntry(n Node) (e *Entry) {
 		}
 		e.Prefix = getRootPrefix(e)
 		addExtraKeywordsToLeafEntry(n, e)
+		e.Description = ""
+		e.PerfOpts = &PerformanceOpts{}
 		return e
 	case *Uses:
 		g := FindGrouping(s, s.Name, map[string]bool{})
@@ -1002,6 +1029,11 @@ func ToEntry(n Node) (e *Entry) {
 		e.Prefix = getRootPrefix(e)
 	}
 
+	// unset for optimization
+	e.Description = ""
+	//e.Extra = nil
+	e.Augmented = nil
+
 	return e
 }
 
@@ -1093,7 +1125,7 @@ func (e *Entry) ApplyDeviate() []error {
 					}
 
 					if devSpec.Default != "" {
-						deviatedNode.Default = ""
+						deviatedNode.Default = devSpec.Default
 					}
 
 					if devSpec.Mandatory != TSUnset {
@@ -1192,15 +1224,17 @@ func (e *Entry) FixChoice() {
 						Source:     ce.Node.Statement(),
 						Extensions: ce.Node.Exts(),
 					},
-					Name:   ce.Name,
-					Kind:   CaseEntry,
-					Config: ce.Config,
-					Prefix: ce.Prefix,
-					Dir:    map[string]*Entry{ce.Name: ce},
-					Extra:  map[string][]interface{}{},
+					Name:     ce.Name,
+					Kind:     CaseEntry,
+					Config:   ce.Config,
+					Prefix:   ce.Prefix,
+					Dir:      map[string]*Entry{ce.Name: ce},
+					Extra:    map[string][]interface{}{},
+					PerfOpts: &PerformanceOpts{},
 				}
 				ce.Parent = ne
 				e.Dir[k] = ne
+				e.DirOKeys = append(e.DirOKeys, k)
 			}
 		}
 	}
@@ -1378,12 +1412,24 @@ func (e *Entry) shallowDup() *Entry {
 	// copied we will have to explicitly uncopy them.
 	ne := *e
 
+	//Copy the ordered Dir keys to new entry
+	if len(e.DirOKeys) > 0 {
+		ne.DirOKeys = make([]string, 0)
+		for _, key := range e.DirOKeys {
+			ne.DirOKeys = append(ne.DirOKeys, key)
+		}
+	}
+
 	// Now only copy direct children, clear their Dir, and fix up
 	// Parent pointers.
 	if e.Dir != nil {
 		ne.Dir = make(map[string]*Entry, len(e.Dir))
+		ne.PerfOpts = &PerformanceOpts{}
 		for k, v := range e.Dir {
 			de := *v
+			if de.PerfOpts != nil {
+				de.PerfOpts = &PerformanceOpts{}
+			}
 			de.Dir = nil
 			de.Parent = &ne
 			ne.Dir[k] = &de
@@ -1400,6 +1446,15 @@ func (e *Entry) dup() *Entry {
 	// such as Exts, Choice and Case, but it is not clear that we need
 	// to do that.
 	ne := *e
+	ne.PerfOpts = &PerformanceOpts{}
+
+	//Copy the ordered Dir keys to new entry
+	if len(e.DirOKeys) > 0 {
+		ne.DirOKeys = make([]string, 0)
+		for _, key := range e.DirOKeys {
+			ne.DirOKeys = append(ne.DirOKeys, key)
+		}
+	}
 
 	// Now recurse down to all of our children, fixing up Parent
 	// pointers as we go.
@@ -1436,6 +1491,7 @@ func (e *Entry) merge(prefix *Value, namespace *Value, oe *Entry) {
 			v.Parent = e
 			v.Exts = append(v.Exts, oe.Exts...)
 			e.Dir[k] = v
+			e.DirOKeys = append(e.DirOKeys, k)
 		}
 	}
 }
diff --git a/pkg/yang/types.go b/pkg/yang/types.go
index 829e816..ebdc210 100644
--- a/pkg/yang/types.go
+++ b/pkg/yang/types.go
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// This file is changed by Broadcom.
+// Modifications - Copyright 2019 Broadcom. The term Broadcom refers to Broadcom Inc. and/or its subsidiaries.
+
 package yang
 
 // This file implements the functions relating to types and typedefs.
@@ -69,6 +72,12 @@ func (d *typeDictionary) findExternal(n Node, prefix, name string) (*Typedef, er
 	}
 	if td := d.find(root, name); td != nil {
 		return td, nil
+	} else {
+		for _, in := range root.Include {
+			if td := typeDict.find(in.Module, name); td != nil {
+				return td, nil
+			}
+		}
 	}
 	if prefix != "" {
 		name = prefix + ":" + name
diff --git a/yang.go b/yang.go
index b709c0c..676252f 100644
--- a/yang.go
+++ b/yang.go
@@ -58,6 +58,7 @@ import (
 type formatter struct {
 	name  string
 	f     func(io.Writer, []*yang.Entry)
+	utilf func([]string, map[string]*yang.Module)
 	help  string
 	flags *getopt.Set
 }
@@ -208,5 +209,8 @@ Formats:
 		entries[x] = yang.ToEntry(mods[n])
 	}
 
+	if format == "annotate" {
+		formatters[format].utilf(files, mods)
+	}
 	formatters[format].f(os.Stdout, entries)
 }
